streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetricsImpl.java

```diff
@@ -117,7 +118,7 @@ public Sensor addLatencyAndThroughputSensor(String scopeName, String entityName,
 
         // first add the global operation metrics if not yet, with the global tags only
         Sensor parent = metrics.sensor(sensorName(operationName, null), recordingLevel);
-        addLatencyMetrics(scopeName, parent, "all", operationName, tagMap);
+        addLatencyMetrics(scopeName, parent, null, operationName, tagMap(tags != null && tags.length > 1 ? tags[0] : "all", "all"));
```

Comment:
Since these APIs are also exposed to users for them to construct their own metrics, in which case the `tags` may not contain any values, but at the same time we are relying on `entityName` to construct the parent-child metrics hierarchy. So we should handle the construction of the tags here than enforcing the callers to pass in `null` as entity name and with the additional tags constructed inside the function.